libcloudforensics/aws.py

```diff
@@ -526,6 +608,92 @@ def GetAccountInformation(self, info):
       raise KeyError('Key must be one of ["UserId", "Account", "Arn"]')
     return account_information.get(info)
 
+  def _GetBootVolumeConfigByAmi(self, ami, boot_volume_size):
+    """Return a boot volume configuration for a given AMI and boot volume size.
+
+    Args:
+      ami (str): The Amazon Machine Image ID.
+      boot_volume_size (int): Size of the boot volume, in GB.
+
+    Returns:
+      dict: A BlockDeviceMappings configuration for the specified AMI.
+
+    Raises:
+      RuntimeError: If AMI details cannot be found.
+    """
+
+    client = self.ClientApi(EC2_SERVICE)
+    try:
+      image = client.describe_images(ImageIds=[ami])
+    except client.exceptions.ClientError as exception:
+      raise RuntimeError(
+          'Could not find image information for AMI {0:s}: {1:s}'.format(
+              ami, str(exception)))
+
+    # If the call to describe_images was successful, then the API's response
+    # is expected to contain at least one image and its corresponding block
+    # device mappings information.
+    block_device_mapping = image['Images'][0]['BlockDeviceMappings'][0]
+    block_device_mapping['Ebs']['VolumeSize'] = boot_volume_size
+    return block_device_mapping
+
+  @staticmethod
+  def _GetInstanceTypeByCPU(cpu_cores):
+    """Return the instance type for the requested number of  CPU cores.
+
+    Args:
+      cpu_cores (int): The number of requested cores.
+
+    Returns:
+      str: The type of instance that matches the number of cores.
+
+    Raises:
+      ValueError: If the requested amount of cores is unavailable.
+    """
+
+    cpus = [1, 2, 4, 8, 16, 32, 40, 48, 64, 96, 128]
+    instance_names = ['t2.small', 'm4.large', 'm4.xlarge', 'm4.2xlarge',
+                      'm4.4xlarge', 'm5.8xlarge', 'm4.10xlarge', 'm5.12xlarge',
+                      'm4.16xlarge', 'm5.24xlarge', 'x1.32xlarge']
+    try:
+      idx = cpus.index(cpu_cores)
```

Comment:
Maybe replace this by

```python
if cpu_cores not in cpus:
  raise ValueError('Cannot start...')
```

In fact, if there's a 1:1 mapping between "number of cores" and "instance type", isn't it easier to just use a dict for this?

```python
cpu_instance_map = {
    1: 't2.small',
    ...
}
if cpu_cores not in instance_type:
  raise ValueError()
return cpu_instance_map[cpu_cores]
```

Reply:
Ah yeah, brain fart! Good catch, thanks